### DISCUSSION TITLE
2022-03-07 weekly planning

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#1142] [Maintainer discussions]
## About
At the start of each week, we start a discussion covering Mathesar's progress, team member check ins, and general updates. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

## Discussion topics
- [Last Week in Mathesar](https://github.com/centerofci/mathesar/discussions/1142#discussioncomment-2298140)
- [Frontend priorities](https://github.com/centerofci/mathesar/discussions/1142#discussioncomment-2298141)
- [Backend priorities](https://github.com/centerofci/mathesar/discussions/1142#discussioncomment-2298145)
- [Design priorities](https://github.com/centerofci/mathesar/discussions/1142#discussioncomment-2298146)
- [Product priorities](https://github.com/centerofci/mathesar/discussions/1142#discussioncomment-2298147)
- [Comms Assignee schedule](https://github.com/centerofci/mathesar/discussions/1142#discussioncomment-2298149)
- [Sean's check in](https://github.com/centerofci/mathesar/discussions/1142#discussioncomment-2304581)
- [Ghislaine's check in](https://github.com/centerofci/mathesar/discussions/1142#discussioncomment-2306696)
- [Dominykas' check in](https://github.com/centerofci/mathesar/discussions/1142#discussioncomment-2308300)
- [Pavish's check in](https://github.com/centerofci/mathesar/discussions/1142#discussioncomment-2310163)
- [Kriti's check in](https://github.com/centerofci/mathesar/discussions/1142#discussioncomment-2310950)
- [Mukesh's check in](https://github.com/centerofci/mathesar/discussions/1142#discussioncomment-2312926)
- [Brent's check in](https://github.com/centerofci/mathesar/discussions/1142#discussioncomment-2314689)
## Weekly check-in
@centerofci/mathesar-core please post your weekly check-in as a top-level response and edit the discussion topics above to add your check in as a topic. Others are also welcome to respond.

Prompts:
- How did your most recent work go?
- What are you planning to work on soon?
- Do you anticipate any dependencies or blockers that someone else on the team can help you with?

If you'd like to add anything else to your check-in that's not related to the prompts, feel free to do so.

**Comment by kgodey** [comment]
## Last Week in Mathesar

### Feedback Requested
- https://github.com/centerofci/mathesar/discussions/1015
- https://github.com/centerofci/mathesar/discussions/1034
- https://github.com/centerofci/mathesar/discussions/1136
  
### Complete
- **Milestones**:
  - [Working with Tables](https://github.com/centerofci/mathesar/milestone/8) with 43 issues complete
- **Discussions**:
  - https://github.com/centerofci/mathesar/discussions/1119
- **Specs**:
  - https://github.com/centerofci/mathesar-wiki/pull/36
- **Wiki updates**:
  - The [Product Spec for Views](https://wiki.mathesar.org/en/product/specs/2022-01-views) is done and is being reviewed, see the "Feedback Requested" section.
- **Full stack and general work**:
  - https://github.com/centerofci/mathesar/pull/1139
  - https://github.com/centerofci/mathesar/pull/1117
- **Backend work**:
  - https://github.com/centerofci/mathesar/issues/388
  - https://github.com/centerofci/mathesar/issues/406
  - https://github.com/centerofci/mathesar/issues/410
  - https://github.com/centerofci/mathesar/issues/413
  - https://github.com/centerofci/mathesar/issues/418
  - https://github.com/centerofci/mathesar/issues/427
  - https://github.com/centerofci/mathesar/issues/431
  - https://github.com/centerofci/mathesar/issues/992
  - https://github.com/centerofci/mathesar/issues/993
  - https://github.com/centerofci/mathesar/issues/995
  - https://github.com/centerofci/mathesar/issues/1092
- **Frontend work**:
  - https://github.com/centerofci/mathesar/issues/254
  - https://github.com/centerofci/mathesar/issues/327
  - https://github.com/centerofci/mathesar/issues/499
  - https://github.com/centerofci/mathesar/issues/928
  - https://github.com/centerofci/mathesar/issues/1113
  - https://github.com/centerofci/mathesar/pull/1109

**Comment by kgodey** [comment]
## Frontend priorities

1. Complete front end changes to https://github.com/centerofci/mathesar/pull/1047
    1. PR: #1141
2. Get open PRs merged:
    1. #1076
    1. #1133
3. Make progress on issues to lay groundwork for [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11) milestone:
    1. #970
    1. #930 
    1. #1108
    1. #1118
    1. #257 
4. Continue working on other [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11) frontend issues.

@pavish @seancolsen Please comment below with thoughts/concerns.

**Reply by seancolsen** [reply]
Looks good.

#970, #930, and #1108 are all tickets I've been planning to work on. I'm planning to raise some questions about #970 before I begin work, and I'm hoping that #930 and #1108 will both be fairly quick. So I'll add a few more tickets to the list, as possible tickets I'd work on this week, if I'm out of un-blocked work with the above list:

- #987
- #982
- #1099

**Reply by seancolsen** [reply]
Now that @pavish has [clarified](https://matrix.to/#/!UnujZDUxGuMrYdvgTU:matrix.mathesar.org/$xrFTHJ31mzaQwK-Yd0LBS1Fnktd33KwEgn1MsmDn9oU?via=matrix.mathesar.org&via=matrix.org) that #775 is in fact not blocked, I'll be adding that near the top of my list. My list is:

- #970 (though I need to have some clarifying conversations before starting work)
- #775
- #930
- #1108
- #987

Not saying I'll get all that done this week though. We'll see how far I get.

**Reply by pavish** [reply]
Looks good to me.

**Comment by kgodey** [comment]
## Backend priorities

The frontend is catching up fast to the backend, so we really need to get backend work for  completed and start on [Links between Tables](https://github.com/centerofci/mathesar/milestone/47) ASAP.

Remaining issues:

1. Get open PRs merged:
    1. https://github.com/centerofci/mathesar/pull/1047
    1. https://github.com/centerofci/mathesar/pull/1093
    1. https://github.com/centerofci/mathesar/pull/1132
    1. https://github.com/centerofci/mathesar/pull/1137
2. Complete remaining backend work for [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11):
    1. @dmos62:
        1. https://github.com/centerofci/mathesar/issues/1097
        1. https://github.com/centerofci/mathesar/issues/1100
        1. https://github.com/centerofci/mathesar/issues/1036
        1. https://github.com/centerofci/mathesar/issues/1037
    2. @mathemancer
        1. https://github.com/centerofci/mathesar/issues/845
        2. **Grouping**:
            1. https://github.com/centerofci/mathesar/issues/408 
            2. https://github.com/centerofci/mathesar/issues/411 
            3. https://github.com/centerofci/mathesar/issues/414
            4. https://github.com/centerofci/mathesar/issues/384
            5. https://github.com/centerofci/mathesar/issues/429
            6. https://github.com/centerofci/mathesar/issues/419
            7. https://github.com/centerofci/mathesar/issues/432
    3. @silentninja:
        1. https://github.com/centerofci/mathesar/issues/839
            1. https://github.com/centerofci/mathesar/issues/994
        1. https://github.com/centerofci/mathesar/issues/996
        1. https://github.com/centerofci/mathesar/issues/1064
        1. https://github.com/centerofci/mathesar/issues/1105
    4. Unassigned
        1. https://github.com/centerofci/mathesar/issues/1121
3. Start work on backend work for [Links between Tables](https://github.com/centerofci/mathesar/milestone/47), some ideas:
    1. https://github.com/centerofci/mathesar/issues/943
    2. https://github.com/centerofci/mathesar/issues/944
    3. https://github.com/centerofci/mathesar/issues/973
    4. https://github.com/centerofci/mathesar/issues/972
4. I should have time to start on #1008 this week.

@dmos62 @mathemancer @silentninja Please comment below with thoughts/concerns.

**Reply by dmos62** [reply]
Sounds good. My tasks are pretty clear cut. URI scheme suggestions are still slightly fuzzy, but I'll figure it out.

**Reply by mathemancer** [reply]
The priorities make sense. I think there are dependencies between us for most current issues, but we'll work through it.

**Comment by kgodey** [comment]
## Design priorities

1. Complete wireframing and draft a spec for #1065
3. Design discussions for #978 

@ghislaineguerin @ppii775 Please comment below with thoughts/concerns

**Reply by ghislaineguerin** [reply]
@kgodey This looks good. As discussed, the plan is to start writing specs for #1065 tomorrow. As for #978 we discussed updates to the issue regarding the examples and scenarios so that @ppii775 can incorporate them into the design.

**Reply by kgodey** [reply]
Sounds good @ghislaineguerin. I did add some examples to #978 in the comments so we should be good to go there.

**Comment by kgodey** [comment]
## Product priorities

1. Finalize [product spec for Views](https://wiki.mathesar.org/en/product/specs/2022-01-views) based on feedback from discussion.

**Comment by kgodey** [comment]
## Comms Assignee schedule

| Week | Person |
|-|-|
| 2022-03-07 to 2022-03-11 | Kriti |
| 2022-03-14 to 2022-03-18 | Pavish |
| 2022-03-21 to 2022-03-25 | Mukesh |
| 2022-03-28 to 2022-04-01 | Sean |
| 2022-04-04 to 2022-04-08 | Dominykas |
| 2022-04-11 to 2022-04-15 | Ghislaine |
| 2022-04-18 to 2022-04-22 | Brent |

### Instructions
You can also find the schedule and instructions for the Comms Assignee on the [wiki page](https://wiki.mathesar.org/team/guide/comms-assignee). Please feel free to update the wiki page if you think of anything useful during your shift.

**Comment by seancolsen** [comment]
## Sean's check in

> How did your most recent work go?

- Last week went well. I'm happy about my progress.

> What are you planning to work on soon?

- See https://github.com/centerofci/mathesar/discussions/1142#discussioncomment-2304580

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

- Nope

**Comment by ghislaineguerin** [comment]
> How did your most recent work go?

It went well. Product discussions were helpful.

> What are you planning to work on soon?

See https://github.com/centerofci/mathesar/discussions/1142#discussioncomment-2306620

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

- Not for now.

**Comment by dmos62** [comment]
>    How did your most recent work go?

Jumped into coding solutions for the URI scheme recommendation issue too early. Ended up repeatedly discarding work, because didn't explore the issue beforehand enough.

>    What are you planning to work on soon?

The task list in the backend priorities post is accurate. I'll start the week with the URI scheme recommendations. I've also a PR review lined up.

>    Do you anticipate any dependencies or blockers that someone else on the team can help you with?

No.

**Comment by pavish** [comment]
## Pavish's check in

> How did your most recent work go?

Pretty well. Most groundwork for type implementation on the frontend is complete.

>  What are you planning to work on soon?

* A draft PR for text type updates is raised: https://github.com/centerofci/mathesar/pull/1144. I'm planning to get it ready first.
* Then I'd work on https://github.com/centerofci/mathesar/issues/257, which includes groundwork for display options for all types. Once that is ready, we can work in parallel on types on the frontend.
* If I get time, I'll start working on implementing another type. I'll decide which ones when I'm ready to begin.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

None

**Comment by kgodey** [comment]
## Kriti's check in

> How did your most recent work go?

Good, the Views spec is mostly done. It took much longer than I thought it would and I'm glad to be moving on to other work. I do think it's going to be great when implemented.

> What are you planning to work on soon?

- Updating the Views spec to its final form based on feedback from #1015 
- Updating design tickets for the Views milestone to match the product spec.
- Starting work on #1008 
- Comms around Google Summer of Code

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

If anyone has feedback about the Views spec, please comment on #1015 in the next day or so. I'm considering the spec closed after that.

**Comment by silentninja** [comment]
## Mukesh's check in

>    How did your most recent work go?

It went well. Got most of the column index replacement PRs merged. 

>    What are you planning to work on soon?

Will be working as per the backend priorities list. 

>    Do you anticipate any dependencies or blockers that someone else on the team can help you with?

I have started a discussion to finalize on the [identifier for DML operations](https://github.com/centerofci/mathesar/discussions/1147). It is preventing the backend work on [https://github.com/centerofci/mathesar/pull/1047](https://github.com/centerofci/mathesar/pull/1047)

**Comment by mathemancer** [comment]
> How did your most recent work go?

It went well, though the custom money type was more work than I originally thought. 

> What are you planning to work on soon?

I'm finishing up the date/time spec, getting my money type PR merged, then getting into grouping. Alongside that, I (we) will be managing an influx of GSoC-related contributors, hopefully.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

Grouping has a dependency on the column ID work @silentninja is doing, along with the resolution to some discussions.
